zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplinePlugin.kt

```diff
@@ -39,14 +40,39 @@ class ZiplinePlugin : KotlinCompilerPluginSupportPlugin {
     version = BuildConfig.KOTLIN_PLUGIN_VERSION,
   )
 
+  override fun apply(target: Project) {
+    super.apply(target)
+
+    // If the build has compileDevelopmentExecutableKotlinJs, create developmentExecutableZipline.
+    // Symmetrically for production.
+    target.tasks.whenTaskAdded { task ->
+      if (task !is KotlinJsCompile) {
+        return@whenTaskAdded
+      }
+      val compileType = when (task.name) {
+        "compileDevelopmentExecutableKotlinJs" -> "developmentExecutable"
+        "compileProductionExecutableKotlinJs" -> "productionExecutable"
+        else -> return@whenTaskAdded
+      }
```

Comment:
This does not support a single JS target with a custom name or multiple JS targets. We will need to iterate over `kotlin.targets`, filter out those which are JS, and create a task associated with each target using its name.

Will need tests which do
```groovy
kotlin {
  js("customName")
}
```
and
```groovy
kotlin {
  js("foo")
  js("bar")
}
```
to confirm behavior.

Reply:
Oooooh that’s what I was looking for. Good call.